table/iterator.cc

```diff
@@ -19,10 +19,18 @@ Cleanable::Cleanable() {
   cleanup_.next = nullptr;
 }
 
-Cleanable::~Cleanable() {
+Cleanable::~Cleanable() { DoCleanup(); }
+
+void Cleanable::Reset() {
```

Comment:
You are right but since Cleanable is inherited in many objects, any small increase in construction/deconstruction cost could potentially be observable. I am more inclined to keep the cost the same as it was before.